File: 0283-move-zeroes/0283-move-zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    let l=0;
    const n = nums.length;
    for (let r=0;r<n;r++){
        if(nums[r] !== 0){
            ;[nums[r],nums[l]]=[nums[l],nums[r]];
            l=l+1;
        }
    }
    return nums;
    
};

File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        r=0
        l=0
        n=len(nums)
        while r<n:
            if nums[r]!=0:
                nums[r],nums[l]=nums[l],nums[r]
                l=l+1
            r=r+1
        return nums
        
        